### DISCUSSION TITLE
How would adoption of the TC39 Record/Tuple proposal affect Solid?

### DISCUSSION THREAD

**Discussion by kee-oth** [solidjs/solid#441] [General]
This maybe be only something for the Solid creator to chime in on but I'm curious how, if the proposal is adopted, JS Records and Tuples (deeply immutable data structures which are composed from primitives and are themselves primitives) could be utilized to increase the performance of Solid. Would change detection operations and memory usage be lighter/faster/less resource intensive to a significant degree?

Here's the proposal:
https://github.com/tc39/proposal-record-tuple/

And here's an article talking about Records and Tuples and how they would affect React (positively!):
https://sebastienlorber.com/records-and-tuples-for-react

Thanks!

**Comment by ryansolid** [comment]
I won't know until I benchmark it. I suspect things might not really apply here as our internals for nested stuff tends to be mutable. I provide immutable interfaces but that is just an API decision.
